frontend/src/__layout.svelte

```diff
@@ -1,11 +1,54 @@
+<script>
+  import Navbar from './lib/Navbar.svelte'; // Import Navbar component
+  import { init, dictionary, getLocaleFromNavigator, locale } from 'svelte-i18n';
+  import en from './i18n/en.json';
+  import fr from './i18n/fr.json';
+  import { t } from 'svelte-i18n';
+
+  let selectedLang;
+
+  if (typeof localStorage !== 'undefined') {
+    selectedLang = localStorage.getItem('language');
+  }
+
+  if (typeof navigator !== 'undefined') {
```

Comment:
naigator will most likely always be defined as an object and never undefined. But perhaps this is for legacy browsers that might not support the navigator object? Doesn't hurt to leave it in, but might be useless.

Reply:
Actually the same applies for localStorage, but again it might help support older browsers.

Reply:
Thanks for the explanations @DenzelBraithwaite , I will leave this in as you suggested.